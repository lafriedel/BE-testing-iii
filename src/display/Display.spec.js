import React from 'react';
import { render, cleanup } from 'react-testing-library';
import renderer from 'react-test-renderer';
import 'jest-dom/extend-expect';
import Display from './Display';

afterEach(cleanup);

describe('Display', () => {
    // test.todo('defaults gate to unlocked and open');

    // test.todo('displays "Closed" if props.closed === true');

    // test.todo('displays "Locked" if props.locked === true');

    // test.todo('uses "red-led" class when displaying "Closed"');

    // test.todo('uses "red-led" class when displaying "Locked"');

    // test.todo('uses "green-led" class when displaying "Open"');

    // test.todo('uses "green-led" class when displaying "Unlocked"');

    test.skip('matches snapshot', () => {
        const tree = renderer.create(<Display />);

        expect(tree).toMatchSnapshot();
    })
})
