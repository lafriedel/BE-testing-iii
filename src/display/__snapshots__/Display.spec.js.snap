// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Display matches snapshot 1`] = `
<div
  className="display panel"
>
  <div
    className="led green-led"
    data-testid="lockStatus"
  >
    Unlocked
  </div>
  <div
    className="led green-led"
    data-testid="openStatus"
  >
    Open
  </div>
</div>
`;
